drivers/bridge/bridge.go

```diff
@@ -432,6 +432,45 @@ func parseNetworkOptions(option options.Generic) (*networkConfiguration, error)
 	return config, nil
 }
 
+// Returns the non link-local IPv6 subnet for the containers attached to this bridge if found, nil otherwise
+func getV6Network(config *networkConfiguration, i *bridgeInterface) *net.IPNet {
+	if config.FixedCIDRv6 != nil {
+		return config.FixedCIDRv6
+	}
+
+	if i.bridgeIPv6 != nil && i.bridgeIPv6.IP != nil && !i.bridgeIPv6.IP.IsLinkLocalUnicast() {
+		return i.bridgeIPv6
+	}
+
+	return nil
+}
+
+func isolateNetworks(d *driver, thisV4, thisV6 *net.IPNet, enable bool) error {
+	d.Lock()
+	defer d.Unlock()
```

Comment:
@aboch agreed that maps are not atomic so we need to protect the maps itself. But I am still against holding `d.Lock` while calling `setINC`. So what you can do is when you are iterating networks you can collect a list of `otherV4` and `otherV6` while holding the `d.Lock` and then unlock and then iterate over a list of `otherV4` and `otherV6` and call `setINC` for each of them. Does that work?